src/main/kotlin/com/manimdsl/runtime/ASTExecutor.kt

```diff
@@ -667,6 +681,60 @@ class VirtualMachine(
             )
         }
 
+
+        private fun executeWhileStatement(whileStatementNode: WhileStatementNode): ExecValue {
+            if (showMoveToLine) addSleep(0.5)
+
+            var conditionValue: ExecValue
+            var execValue: ExecValue
+            var loopCount = 0
+
+            while (loopCount < MAX_NUMBER_OF_LOOPS) {
```

Comment:
@mayanksharma3 @adityagoel4512 does this look better?

Reply:
Yeah I think it works, it makes sense that the continue and break are done at the bottom with return and continue, and there is a limit check now so I'm happy with this